rust/nasl-builtin-regex/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+use nasl_builtin_utils::{error::FunctionErrorKind, NaslFunction};
+use nasl_builtin_utils::{Context, Register};
+use nasl_function_proc_macro::nasl_function;
+use nasl_syntax::NaslValue;
+use regex::RegexBuilder;
+
+/// Matches a string against a regular expression.
+/// - string  String to search the pattern in
+/// - pattern the pattern that should be matched
+/// - icase   case insensitive flag
+/// - rnul    replace the null char in the string. Default TRUE.
+/// - multiline Is FALSE by default (string is truncated at the first
+///                “end of line”), and can be set to TRUE for multiline search.
+/// Return true if matches, false otherwise
+#[nasl_function(named(string, pattern, icase, rnul, multiline))]
+fn ereg(
+    string: NaslValue,
+    pattern: NaslValue,
+    icase: Option<bool>,
+    rnul: Option<bool>,
+    multiline: Option<bool>,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let icase = icase.unwrap_or_default();
+    let rnul = rnul.unwrap_or(true);
+    let multiline = multiline.unwrap_or_default();
+
+    let str_bind = string.to_string();
+    let mut str_aux = str_bind.as_str();
+
+    if !rnul {
+        let s = match str_aux.chars().position(|c| c == '\0') {
+            Some(i) => i,
+            None => str_aux.len(),
+        };
+        (str_aux, _) = str_aux.split_at(s);
+    }
+
+    if !multiline {
+        let s = match str_aux.chars().position(|c| c == '\n') {
+            Some(i) => i,
+            None => str_aux.len(),
+        };
+        (str_aux, _) = str_aux.split_at(s);
+    }
```

Comment:
```suggestion
    if !rnul {
        str_aux = str_aux.split('\0').next().unwrap();
    }
    if !multiline {
        str_aux = str_aux.split('\n').next().unwrap();
    }
```

Reply:
Also, this logic appears many times throughout the code. Please extract it into a function:

```rust
fn parse_search_string(s: &str, rnul: bool, multiline: bool) -> &str {
    ...
}
```